src/debuglog.js

```diff
@@ -42,19 +42,19 @@ function debuglog (set) {
 	if (debugEnv === undefined) {
 		debugEnv = process.env.NODE_DEBUG || ''
 	}
-	set = set.toUpperCase()
-	if (!debugs[set]) {
-		if (debugEnv === '*' || new RegExp('\\b' + set + '\\b', 'i').test(debugEnv)) {
+	const upperSet = set.toUpperCase()
```

Comment:
@alexanderGugel, I'm fine with reverting it, issue was [no-param-reassign](http://eslint.org/docs/rules/no-param-reassign), but I've added exceptions elsewhere, so I will do that as well.